vendor/github.com/docker/distribution/registry/client/auth/challenge/authchallenge.go

```diff
@@ -86,6 +88,24 @@ func (m *simpleManager) AddResponse(resp *http.Response) error {
 	return nil
 }
 
+func (m *simpleManager) GetHost() string {
```

Comment:
We can't just edit a file from the vendored dependency. If a change is needed here, it would need to get into a https://github.com/distribution/distribution release first. Then we could revendor the `distribution/distribution` to use it.

Reply:
Instead of adding this function, perhaps it would be possible to get the same data inside the `PingV2Registry` function and then just return the realm host?

Reply:
OK taken in account. Done in PingV2Registry function